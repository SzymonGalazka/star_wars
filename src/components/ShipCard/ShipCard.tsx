import React from "react";
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

import { ShipDataType } from '../../utils/playerModels';

const useStyles = makeStyles({
  root: {
    marginLeft: 10,
  },
  title: {
    fontSize: 18,
  },
  pos: {
    marginBottom: 12,
  },
  winner: {
    color: '#2E8B57',
  },
});

const ShipCard = (props: ShipDataType & { isWinner: boolean }) => {
  const classes = useStyles();

  return (
    <div style={{ display: "inline-block", margin: "10px" }}>
      <Card className={classes.root}>
        <CardContent>
          <Typography className={classes.title} color="textSecondary" data-testid="ship_card_name">
            <b>Name: </b> {props.name}
          </Typography>
          <Typography className={classes.pos} color="textSecondary" data-testid="ship_card_model">
            <b>Model: </b> {props.model}
          </Typography>
          <Typography className={classes.pos} color="textSecondary" data-testid="ship_card_crew">
            <b>{props.crew} </b> crew member(s)
            </Typography>
          {props.isWinner && <Typography className={classes.winner} color="textSecondary" data-testid="ship_card_winner">
            <b> Wins! </b>
          </Typography>}
        </CardContent>
      </Card>
    </div>
  );
}

export default ShipCard;