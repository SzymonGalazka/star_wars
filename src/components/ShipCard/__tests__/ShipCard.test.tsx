import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, getByTestId } from "@testing-library/react";

import ShipCard from '../ShipCard';

describe("ShipCard", () => {
  it("should render ShipCard component", async () => {
    const tree = TestRenderer.create(<ShipCard name="testShipName" model="testShipModel" crew={34} isWinner={false} />);
    expect(tree).toMatchSnapshot();
  });

  it("should have name displayed", async () => {
    const { container } = render(
      <ShipCard name="testShipName" model="testShipModel" crew={34} isWinner={false} />
    );
    const element = getByTestId(container, 'ship_card_name')
    expect(element).toHaveTextContent(
      "testShipName"
    );
  });

  it("should have model displayed", async () => {
    const { container } = render(
      <ShipCard name="testShipName" model="testShipModel" crew={34} isWinner={false} />
    );
    const element = getByTestId(container, 'ship_card_model')
    expect(element).toHaveTextContent(
      "testShipModel"
    );
  });

  it("should have crew number displayed", async () => {
    const { container } = render(
      <ShipCard name="testShipName" model="testShipModel" crew={34} isWinner={false} />
    );
    const element = getByTestId(container, 'ship_card_crew')

    expect(element).toHaveTextContent(
      "34"
    );
  });

  it("should display winner section", async () => {
    const { container } = render(
      <ShipCard name="testShipName" model="testShipModel" crew={34} isWinner={true} />
    );
    const element = getByTestId(container, 'ship_card_winner')
    expect(element).toHaveTextContent(
      "Wins!"
    );

  });
  
});