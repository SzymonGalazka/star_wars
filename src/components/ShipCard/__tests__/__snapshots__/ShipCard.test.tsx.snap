// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipCard should render ShipCard component 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "margin": "10px",
    }
  }
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-title-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
        data-testid="ship_card_name"
      >
        <b>
          Name: 
        </b>
         
        testShipName
      </p>
      <p
        className="MuiTypography-root makeStyles-pos-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
        data-testid="ship_card_model"
      >
        <b>
          Model: 
        </b>
         
        testShipModel
      </p>
      <p
        className="MuiTypography-root makeStyles-pos-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
        data-testid="ship_card_crew"
      >
        <b>
          34
           
        </b>
         crew member(s)
      </p>
    </div>
  </div>
</div>
`;
